pkg/build/builder/source.go

```diff
@@ -77,7 +77,9 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
 func checkRemoteGit(url string, timeout time.Duration) error {
 	glog.V(4).Infof("git ls-remote --heads %q", url)
 	cmd := exec.Command("git", "ls-remote", "--heads", url)
-	cmd.Env = []string{"GIT_ASKPASS=/bin/true", "GIT_SSH=" + os.Getenv("GIT_SSH")}
+	env := os.Environ()
+	env = append(env, "GIT_ASKPASS=/bin/true")
```

Comment:
@csrwng - your change looks good wrt the fix I introduced with https://github.com/openshift/origin/commit/59c7fab89d04e25b449283c64bc41885b1f9b654#diff-986cfe99352ecdaac7e8c0d4e47bdbd4 - just picking up any additional relevant env vars.

Couldn't tell if this point was understood in the above discussion (so I'll reiterate it explicitly now just in case):  the original setting of GIT_SSH done [here](https://github.com/openshift/origin/blob/master/pkg/build/builder/cmd/scmauth/sshkey.go#L32) via os.Setenv was not sufficient for the golang fork/exec of the `git`.  The child process did *NOT* pick up the dynamically set GIT_SSH env var.